run/hello-broken/e2e_test.py

```diff
@@ -23,86 +23,121 @@
 
 import pytest
 
+# Unique suffix to create distinct service names
+SUFFIX = uuid.uuid4().hex[:10]
+PROJECT = os.environ["GOOGLE_CLOUD_PROJECT"]
+IMAGE_NAME = f"gcr.io/{PROJECT}/hello-{SUFFIX}"
 
-@pytest.fixture()
-def services():
-    # Unique suffix to create distinct service names
-    suffix = uuid.uuid4().hex
-    project = os.environ["GOOGLE_CLOUD_PROJECT"]
 
-    # Build and Deploy Cloud Run Services
+@pytest.fixture
+def container_image():
+    # Build container image for Cloud Run deployment
     subprocess.run(
         [
             "gcloud",
             "builds",
             "submit",
+            "--tag",
+            IMAGE_NAME,
             "--project",
-            project,
-            "--substitutions",
-            f"_SUFFIX={suffix}",
-            "--config",
-            "e2e_test_setup.yaml",
+            PROJECT,
             "--quiet",
-        ],
-        check=True,
+        ], check=True
     )
+    
+    yield IMAGE_NAME
 
-    # Get the URL for the service and the token
-    service = subprocess.run(
+    # Delete container image
+    subprocess.run(
         [
             "gcloud",
-            "run",
+            "container",
+            "images",
+            "delete",
+            IMAGE_NAME,
+            "--quiet",
             "--project",
-            project,
-            "--platform=managed",
+            PROJECT,
+        ], check=True
+    )
+
+@pytest.fixture
+def deployed_service(container_image):
+    # Deploy image to Cloud Run
+    service_name = f"hello-{SUFFIX}"
+    subprocess.run(
```

Comment:
These should all have check=True,  if they fail -- there isn't any point trying the rest of the commands